networklibrary/src/main/java/com/eis0/networklibrary/Network.java

```diff
@@ -0,0 +1,151 @@
+package com.eis0.networklibrary;
+
+import androidx.annotation.NonNull;
+
+import com.eis0.smslibrary.SMSManager;
+import com.eis0.smslibrary.SMSMessage;
+import com.eis0.smslibrary.SMSPeer;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Class that represent a replicated network object. A replicated network is basically a list of
+ * peers on which is possible to execute some message related commands like unicasting,
+ * multicasting and broadcasting. Every node of the network has an exact copy of all the resources
+ * and nodes in the network and so the resource searches are made locally.<br>
+ * A network without peers is said to be a LOCALNET, a network that includes only the local
+ * host.<br>
+ * More formally, every network is composed by the local host and a list of peers, if the latter is
+ * empty, the network became a LOCALNET.<br>
+ * The creation of {@link Network} objects is managed by the {@link NetworksPool}.
+ *
+ * @author Matteo Carnelos
+ */
+public class Network {
+
+    // Every network is addressed by the combinations of all peers, but for the LOCALNET, where the
+    // are no peers, is given a special address
+    public final static String LOCALNET_ADDR = "localnet";
```

Comment:
### LOCALNET Strategy Explanation
I needed a way to represent a `Network` with only this device.
The simplest idea was to create a `Network` with only the peer of the device, but obtaining the phone address of a device is oddly difficult, there are basically two ways:
1. Using Android APIs. There is an API that returns the phone address of the device, but it's not reliable at all: it can return either `null`,`""`, `"??????????"` or, in the worst case, a valid phone number of a previous SIM inserted in the device.
2. Asking the user. This is the preferred way for most of the commercial apps that need the phone number of the device (e.g. WhatsApp), and the only way to verify that the user has inserted his number is to send him a verification code, but for doing this I would need to purchase an SMS server, so I abandoned also this solution.

Finally I came up with the idea of the LOCALNET, that is a `Network` without peers, and it is interpreted as a network with only this device. In this particular type of situation, in fact, knowing my phone number is not needed.
So, when a device needs to address the LOCALNET of another device, it just need to give him the special address I created: the LOCALNET_ADDR, that in this case is the string "localnet". I was widely inspired by the "localhost" address in Ethernet Networks.

This solution requires a bit more behind-the-scenes work but for the user is invisible and it works just fine.